### PR TITLE
[Fix] Clamp degenerate all-sentinel draft rows to token 0 in dspark `_online_combine_kernel`

### PR DESCRIPTION
**Pull request by hnyls2002**
`_online_combine_kernel` picks the next draft token as the min candidate index among tied maxima. If a step-logits row is degenerate (e.g. all `-inf`, so no `rescaled == best` comparison is true), `cand` stays all-`_IDX_SENTINEL` (2147483647) and that sentinel is written as a token id, which then asserts out-of-range in the downstream embedding `index_select`.

Guard by clamping an all-sentinel row to token 0 so a valid id is always emitted; a degenerate draft is harmless (rejected at verify). Defense-in-depth, independent of any upstream cause of the degenerate row.

---
### CI States

Latest PR Test (Base): :x: [Run #30069201590](https://github.com/sgl-project/sglang/actions/runs/30069201590)
Latest PR Test (Extra): :x: [Run #30069201462](https://github.com/sgl-project/sglang/actions/runs/30069201462)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test test/registered/spec/dspark/test_dspark_kernel_parity.py

**Comment by hnyls2002** [comment]
/rerun-test test_dspark_kernel_parity.py test_basic_sanity_dspark.py
